js/src/ExpectedMark.js

```diff
@@ -7,7 +7,11 @@ const _expectedMarks = [];
  * Class describes expected marks
  * These marks that have to be recorded before zone is considered complete
  */
-export default class ExpectedMark {
+export default class ExpectedMark extends UXBase {
+  // list of zone callbacks to call on completion
+  onMarkListeners = [];
+  name = this.props.name;
```

Comment:
let's skip this 'alias' to keep a single source of 'name' truth

Reply:
`mark.name` is referenced in a few places in - https://github.com/meetup/ux-capture/pull/8/files#diff-1ef3d18dda593c3ce68ba05749eca236R51.

Reply:
unless you mean that we should use `mark.props.name`? That feels odd to me

Reply:
ah, I see. Interestingly, that oddness is probably the result of some awkward data handling that we can refactor to avoid - essentially we're doing something like

1. get the name of the expected mark
2. create an ExpectedMark with that name
3. listen to the ExpectedMark's `onMark`
4. read the name of the ExpectedMark from the arg to the callback

The name that is returned in (4) is the same name we provided in (1), so we don't actually need it.

Fine to `mark.name` assignment as-is for now, but I might try my hand at making that event handling more streamlined

Reply:
i'll take a stab at this 👍

Reply:
ah, i think i see how to go about this. it's going to be a slightly bigger diff than i was thinking so i'll do it as a follow on right after merging this, since you also have a ticket blocked by this.